great_expectations/checkpoint/checkpoint.py

```diff
@@ -492,79 +419,171 @@ def self_check(self, pretty_print=True) -> dict:
     # noinspection PyShadowingBuiltins
     def get_config(
         self,
-        reconcile: bool = False,
-        runtime_kwargs: Optional[dict] = None,
-        format: str = "dict",
+        mode: str = "typed",
         clean_falsy: bool = False,
-    ) -> Union[dict, str]:
-        if reconcile or runtime_kwargs:
-            config_kwargs: dict = self.get_substituted_config(
-                runtime_kwargs=runtime_kwargs
-            )
-        else:
-            config_kwargs = copy.deepcopy(self.config_kwargs)
+    ) -> Union[CheckpointConfig, dict, str]:
+        config: CheckpointConfig = self.checkpoint_config
+
+        if mode == "typed":
+            return config
+
+        if mode == "commented_map":
+            return config.commented_map
+
+        if mode == "dict":
+            config_kwargs: dict = config.to_dict()
+            if clean_falsy:
+                filter_properties_dict(
+                    properties=config_kwargs,
+                    clean_falsy=True,
+                    inplace=True,
+                )
 
-        if clean_falsy:
-            filter_properties_dict(
-                properties=config_kwargs,
-                clean_falsy=True,
-                inplace=True,
-            )
+            return config_kwargs
 
-        if format == "dict":
-            return dict(**config_kwargs)
+        if mode == "json_dict":
+            config_kwargs: dict = config.to_json_dict()
+            if clean_falsy:
+                filter_properties_dict(
+                    properties=config_kwargs,
+                    clean_falsy=True,
+                    inplace=True,
+                )
 
-        if format in ["str", "dir", "repr"]:
-            json_dict: dict = convert_to_json_serializable(data=config_kwargs)
-            deep_filter_properties_iterable(
-                properties=json_dict,
-                inplace=True,
-            )
-            return json.dumps(json_dict, indent=2)
+            return config_kwargs
 
-        if format == "yaml":
-            return object_to_yaml_str(obj=CommentedMap(**config_kwargs))
+        if mode == "yaml":
+            return config.to_yaml_str()
 
-        raise ValueError(f"Unknown format {format} in LegacyCheckpoint.get_config.")
+        raise ValueError(f'Unknown mode {mode} in "CheckpointBase.get_config()".')
 
     @property
-    def config_kwargs(self) -> Attributes:
-        config_kwargs: Attributes = self._convert_dictionaries_to_attributes(
-            source=self._config_kwargs
-        )
-        return config_kwargs
+    def checkpoint_config(self) -> CheckpointConfig:
+        return self._checkpoint_config
+
+    @checkpoint_config.setter
+    def checkpoint_config(self, value: CheckpointConfig):
+        self._checkpoint_config = value
```

Comment:
@cdkini @donaldheppner Please see my (longer) answer above -- since the configuration object is immutable and resides in the `Base` object (see `DataContextBase`), we have the benefit of carrying it for convenience.  Thanks.